src/mixins/labels.js

```diff
@@ -0,0 +1,115 @@
+import * as THREE from "three";
+
+import { ATOM_GROUP_NAME } from "../enums";
+
+/*
+ * Mixin containing the logic for dealing with atom labes.
+ * Dynamically draws labels over atoms.
+ */
+export const LabelsMixin = (superclass) =>
+    class extends superclass {
+        // eslint-disable-next-line class-methods-use-this
+        createTextTexture(text) {
+            const {
+                fontFace,
+                fontSize,
+                fontWeight,
+                fillColor,
+                strokeColor,
+                strokeLineWidth,
+                textAlign,
+                textBaseline,
+            } = this.settings.labels;
+            const canvas = document.createElement("canvas");
+            const context = canvas.getContext("2d");
+            context.font = `${fontWeight} ${fontSize}px ${fontFace}`;
+            const metrics = context.measureText(text);
+            const textWidth = metrics.width;
+            const basicTextSize = textWidth > fontSize ? textWidth : fontSize;
+            const textSizePowOf2 = 2 ** Math.floor(Math.log2(basicTextSize));
+            const scaledFontSize = (fontSize * textSizePowOf2) / basicTextSize;
+            canvas.width = textSizePowOf2;
+            canvas.height = textSizePowOf2;
+            context.font = `${fontWeight} ${scaledFontSize}px ${fontFace}`;
+            context.textAlign = textAlign;
+            context.textBaseline = textBaseline;
+            context.fillStyle = fillColor;
```

Comment:
maybe let's just use `fillStyle` in settings?

Reply:
OK